src/python/pants/notes/2.9.x.md

```diff
@@ -0,0 +1,105 @@
+# 2.9.x Release Series
+
+## 2.9.0.dev0 (Nov 02, 2021)
+
+### New Features
+
+* Add overrides to python_requirements and poetry_requirements ([#13458](https://github.com/pantsbuild/pants/pull/13458))
+
+* Add `--tailor-check` ([#13432](https://github.com/pantsbuild/pants/pull/13432))
+
+* An option to not clean up the binary chroot for the run goal. ([#13427](https://github.com/pantsbuild/pants/pull/13427))
+
+* Add `[tailor].ignore_paths` and `[tailor].ignore_adding_targets` ([#13422](https://github.com/pantsbuild/pants/pull/13422))
+
+* Validate that `[tailor].build_file_name` is compatible with `[GLOBAL].build_patterns` ([#13420](https://github.com/pantsbuild/pants/pull/13420))
+
+### User API Changes
+
+* Remove deprecated --pants-supportdir. ([#13473](https://github.com/pantsbuild/pants/pull/13473))
+
+* Remove setup_py_commands from python_distribution. ([#13450](https://github.com/pantsbuild/pants/pull/13450))
+
+* Remove deprecated `InferredDependencies.sibling_dependencies_inferrable` kwarg ([#13440](https://github.com/pantsbuild/pants/pull/13440))
+
+* `CheckResults` no longer accepts deprecated `typechecker_name` kwarg ([#13434](https://github.com/pantsbuild/pants/pull/13434))
+
+* Remove deprecated `[pytest].junit_xml_dir` option. ([#13444](https://github.com/pantsbuild/pants/pull/13444))
+
+* Remove deprecated `--list-provides` option ([#13441](https://github.com/pantsbuild/pants/pull/13441))
+
+* Remove deprecated `module_mapping` and `type_stubs_module_mapping` fields ([#13442](https://github.com/pantsbuild/pants/pull/13442))
+
+* Remove `entry_point=<none>` for `pex_binary` in favor of leaving off the field ([#13443](https://github.com/pantsbuild/pants/pull/13443))
+
+* Remove deprecated `Sources` type in favor of `SourcesField` ([#13439](https://github.com/pantsbuild/pants/pull/13439))
+
+* Remove deprecated `requirements_relpath` from `python_requirements` macro ([#13437](https://github.com/pantsbuild/pants/pull/13437))
+
+* Remove deprecated `protobuf_library`, `python_library`, `shell_library`, and `python_requirement_library` target aliases ([#13429](https://github.com/pantsbuild/pants/pull/13429))
+
+* Remove deprecated `--peek-output` ([#13433](https://github.com/pantsbuild/pants/pull/13433))
+
+* Remove dependencies goal deprecated --type option. ([#13431](https://github.com/pantsbuild/pants/pull/13431))
+
+* Remove dependees `--output-format` option. ([#13430](https://github.com/pantsbuild/pants/pull/13430))
+
+* Validate `python_test` and `python_tests` targets not used for `conftest.py` and `test_foo.pyi` ([#13436](https://github.com/pantsbuild/pants/pull/13436))
+
+* Use pants config to specify lockfile locations instead of BUILD targets ([#13295](https://github.com/pantsbuild/pants/pull/13295))
```

Comment:
This is for JVM, so should be left off because it's still so experimental and not documented. 

(We did this with Go too until recently. Now it's stable enough and documented that we include changes)